sound/soc/sof/mediatek/mt8195/mt8195.c

```diff
@@ -0,0 +1,359 @@
+	struct device_node *mem_region;
+	struct device *dev = &pdev->dev;
+	struct adsp_chip_info *adsp = data;
+	int i, ret;
+
+	res = &resource;
+	mem_region = of_parse_phandle(dev->of_node, "memory-region", 0);
+	if (!mem_region) {
+		dev_err(dev, "no 'memory-region' phandle\n");
+		return -ENODEV;
+	}
+
+	ret = of_address_to_resource(mem_region, 0, res);
+	if (ret) {
+		dev_err(dev, "of_address_to_resource failed\n");
+		return ret;
+	}
+
+	dev_info(dev, "[ADSP][DMA] pbase=0x%llx, size=0x%llx\n",
+		 (phys_addr_t)res->start,
+		 resource_size(res));
+
+	ret = of_reserved_mem_device_init(dev);
+	if (ret) {
+		dev_err(dev, "of_reserved_mem_device_init failed\n");
+		return ret;
+	}
+
+	mem_region = of_parse_phandle(dev->of_node, "memory-region", 1);
+	if (!mem_region) {
+		dev_err(dev, "no 'memory-region' phandle\n");
+		return -ENODEV;
+	}
+
+	ret = of_address_to_resource(mem_region, 0, res);
+	if (ret) {
+		dev_err(dev, "of_address_to_resource failed\n");
+		return ret;
+	}
+
+	adsp->pa_dram = (phys_addr_t)res->start;
+	adsp->dramsize = (u32)resource_size(res);
+	if (((u32)adsp->pa_dram) & 0x1000) {
+		dev_err(dev, "adsp memory(0x%x) is not 4K-aligned\n",
+			(u32)adsp->pa_dram);
+		return -EINVAL;
+	}
+
+	if (adsp->dramsize < TOTAL_SIZE_SHARED_DRAM_FROM_TAIL) {
+		dev_err(dev, "adsp memory(0x%x) is not enough for share\n",
+			(u32)adsp->dramsize);
+		return -EINVAL;
+	}
+
+	dev_info(dev, "[ADSP] dram pbase=%pa, dramsize=0x%x\n",
+		 &adsp->pa_dram,
+		 adsp->dramsize);
+
+	/* Parse CFG base */
+	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
+	if (!res) {
+		dev_err(dev, "no ADSP-CFG register resource\n");
+		return -ENXIO;
+	}
+	/* remap for DSP register accessing */
+	adsp->va_cfgreg = devm_ioremap_resource(dev, res);
+	if (IS_ERR(adsp->va_cfgreg))
+		return PTR_ERR(adsp->va_cfgreg);
+
+	adsp->pa_cfgreg = (phys_addr_t)res->start;
+	adsp->cfgregsize = (u32)resource_size(res);
+
+	dev_info(dev, "[ADSP] cfgreg-vbase=%p, cfgregsize=0x%x\n",
+		 adsp->va_cfgreg, adsp->cfgregsize);
+
+	/* Parse SRAM */
+	res = platform_get_resource(pdev, IORESOURCE_MEM, 1);
+	if (!res) {
+		dev_err(dev, "no SRAM resource\n");
+		return -ENXIO;
+	}
+
+	adsp->pa_sram = (phys_addr_t)res->start;
+	adsp->srammsize = (u32)resource_size(res);
+	if (adsp->srammsize < TOTAL_SIZE_SHARED_SRAM_FROM_TAIL) {
+		dev_err(dev, "adsp SRAM(0x%x) is not enough for share\n",
+			(u32)adsp->srammsize);
+		return -EINVAL;
+	}
+
+	dev_info(dev, "[ADSP] sram pbase=%pa,0x%x\n", &adsp->pa_sram,
+		 adsp->srammsize);
+
+	/* Parse MBOX base */
+	for (i = 0; i < DSP_MBOX_NUM; i++) {
+		res = platform_get_resource(pdev, IORESOURCE_MEM, 2 + i);
+		if (!res) {
+			dev_err(dev, "no MBOX resource %d\n", i);
+			return -ENXIO;
+		}
+
+		adsp->pa_mboxreg[i] = (phys_addr_t)res->start;
+		adsp->va_mboxreg[i] = devm_ioremap_resource(dev, res);
+		if (IS_ERR(adsp->va_mboxreg[i]))
+			return PTR_ERR(adsp->va_mboxreg[i]);
+
+		dev_info(dev, "[ADSP]MBOX%d pa_mboxreg:%pa, va_mboxreg:%p\n",
```

Comment:
could you please go through all th dev_info's in your series to make sure they need to be info and not dbg?